ctlcmd/cmdbase.py

```diff
@@ -771,7 +772,176 @@ def globcomp(text):
+  configurations while still have the results be saved in distinct files.
+  Placeholders will be specified in the angle braced. For the full list of
+  special placeholders, as well as how the file parsing is handled, see the
+  detailed documentation in the `ctlcmd.cmdbase.savefilecmd.parse` method.
+  """
+  ##TODO: update this to make the file location customizable
+  DEFAULT_ROOTFILE = 'SAVEROOT_<TIMESTAMP>'
+  def __init__(self, cmd):
+    print("initialize rootfilecmd")
+    controlcmd.__init__(self, cmd)
+    
+  def add_args(self):
+    print("add_args in rootfilecmd")
+    group = self.parser.add_argument_group(
+        "root file saving options", """Options for changing the root file location.
+        For more details, see the official documentation.""")
+    group.add_argument('--saveroot',
+                       type=str,
+                       default=self.DEFAULT_ROOTFILE,
+                       help="""File path to save (placeholders in angle braces).
+                       default=%(default)s""")
+ 
+  def parse(self,args):
+    print("parse in rootfilecmd")
+    if not args.saveroot:  # Early exit if savefile is not set
+      self.saveroot = None
+      print("self.saveroot is None")
+      return args
+    filename = args.saveroot
+
+    # Adding time stamp filenames
+    timestring = datetime.datetime.now().strftime('%Y%m%d_%H%M%S')
+    filename = re.sub('<TIMESTAMP>', timestring, filename, flags=re.IGNORECASE)
 
+    # Adding boardid to the settings
+    filename = re.sub('<BOARDID>',
+                      str(self.board.boardid),
+                      filename,
+                      flags=re.IGNORECASE)
+    # Adding boardtype to the settings
+    filename = re.sub('<BOARDTYPE>',
+                      str(self.board.boardtype),
+                      filename,
+                      flags=re.IGNORECASE)
+
+    # Substituting tokens for argument values
+    for action in self.parser._actions:
+      for string in action.option_strings:
+        string = string.strip('-')
+        if not hasattr(args, string): continue
+        val = ''
+        try:
+          val = action.type(getattr(args, string))
+        except:
+          pass
+        substring = '{0}{1}'.format(string, val)
+        substring = re.sub(r'\.', 'p', substring)
+        filename = re.sub('\\<{0}\\>'.format(string),
+                          substring,
+                          filename,
+                          flags=re.IGNORECASE)
+      self.saveroot = filename 
+      return args
+      
+  def openroot(self,functiontype):
+    ##Make root file
+    file = uproot.recreate(self.saveroot)
+    
+    #Create the titles for the branch names and their data types
+    standard_dict={"time":np.float32,"det_id":int,"gantry x":np.float32,"gantry y":np.float32,
+                   "gantry z":np.float32,"LED bias voltage":np.float32,"LED temp":np.float32,"SiPM temp":np.float32 }
+    halign_dict={"lumival":np.float64,"uncval":np.float64}
```

Comment:
Have these dictionaries be something that is provided by the inherit classes, not something implemented in the base class. We wouldn't want to modify the base class every time a new function is implemented.